### PR TITLE
Update drawer when re-opened via intent

### LINKED ISSUE #3764: Drawer ist not updated if app is re-opened via an intent
Please search to check for an existing issue (including closed issues, for which the fix may not have yet been released) before opening a new issue: https://github.com/k9mail/k-9/issues?q=is%3Aissue

### Expected behavior
When re-opened e.g. from the widget, the drawer content should be updated to reflect possible account changes.

### Actual behavior
The drawer keeps the old account and its folder list.

### Steps to reproduce
1. Open k9
2. Select account 1
3. Close app
4. open message from account 2 via widget
5. open drawer and see that account 1 is still selected

### Environment
K-9 Mail version: 67df429522620dadeaa40e7628a72a6039a0f04c

Android version: 9.0

Account type (IMAP, POP3, WebDAV/Exchange): IMAP

### PR DESCRIPTION
**Pull request by morckx**
Fixes #3764

### CODE REVIEW

**Comment by morckx** [comment]
Do not merge. Has still some unwanted side effects.

**Comment by morckx** [comment]
Seams to work now. Let's make this nicer when more logic is removed from the drawer.

**Comment by morckx** [comment]
Oops, still one issue left: (unified inbox opened when real accout was selected).

**Comment by cketti** [comment]
If viewing the unified inbox and there's only one account `initializeFromLocalSearch()` sets `singleAccountMode` to true and `account` to that account. That doesn't match your expectation that `account` is null if the unified inbox is displayed. The code further down "corrects" the value in `account` but doesn't change `singleAccountMode`. Together this will lead to the following branch being executed.

```java
        if (singleAccountMode && (account == null || !account.isAvailable(this))) {
            Timber.i("not opening MessageList of unavailable account");
            onAccountUnavailable();
            return false;
        }
```

I suggest doing the following:
- extract the code to configure the drawer to a separate method. It shouldn't have any side effects besides configuring the drawer, i.e. certainly not setting `account` to null
- change the "account unavailable" check to `if (account != null && !account.isAvailable(this)) { … }`
- get rid of of the `singleAccountMode` field
- in `initializeFromLocalSearch()`:
  - when `search.searchAllAccounts()` is true set `account` to null and `singleFolderMode` to false
  - only set `singleFolderMode` to true if there's one account and one folder (and `search.searchAllAccounts()` is false)

**Comment by morckx** [comment]
Thanks, that sounds reasonable. I guess I'll have some time tomorrow.

File: app/ui/src/main/java/com/fsck/k9/activity/MessageList.java [hunk 1]
```diff
@@ -1688,31 +1692,32 @@ private void unlockDrawer() {
...
-            singleAccountMode = (accounts.size() == 1);
-            if (singleAccountMode) {
-                account = accounts.get(0);
-            }
+            account = null;
         } else {
             String[] accountUuids = search.getAccountUuids();
-            singleAccountMode = (accountUuids.length == 1);
-            if (singleAccountMode) {
+            if (accountUuids.length == 1) {
```
Comment (cketti) [inline app/ui/src/main/java/com/fsck/k9/activity/MessageList.java]: This doesn't set `account` to null in case the condition isn't met.

Comment (morckx) [reply app/ui/src/main/java/com/fsck/k9/activity/MessageList.java]: I was already wondering under what conditions this situation occurs and what should happen then.

OK, the latter is answered and more important. Will add an else-branch to favour legibility over elegance.

**Pull request merged**